scalableminds/webknossos
# Support float layers

Currently the expected float range is hardcoded to 0-255. With #3583 the range can then be properly configured.

I've also refactored the bucket class to use appropriate TypedArrays instead of Uint8Arrays for everything which makes data handling much easier. However, as a result, volume labeling of uint24 segmentation layers is currently broken (are there uint24 segmentation layers?), because there is not Uint24Array.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Unpack the attached float layer into a dataset folder.
- Open the dataset Edit view and make sure that the elementClass for that layer is set to "float"
- View the dataset, you should be able to see a beautiful image :)

[color_float.zip](https://github.com/scalableminds/webknossos/files/3183145/color_float.zip)

### Issues:
- fixes #3506 

------
~~- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
~~- [ ] Updated [documentation](../blob/master/docs) if applicable~~
~~- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
